tests/smoke_test.py

```diff
@@ -0,0 +1,238 @@
+import argparse
+import datetime
+import json
+import os
+import subprocess
+
+import pandas as pd
+
+from examples.configs.algorithm import algorithm_defaults
+from examples.evaluate import get_metrics
+from wilds import benchmark_datasets
+
+"""
+Runs the smoke tests for WILDS in CodaLab:
+    1. Run ERM on all the benchmarks
+    2. Run reweighted (ERM + uniform_over_groups), IRM, CORAL and GroupDRO on CivilComments
+
+Only small fractions of the datasets are used for testing purposes. This test suite should
+take around 2 hours to finish.
+
+Usage:
+
+    python3 tests/smoke_test.py --wilds-src-uuid <UUID of the WILDS source to test>
+    python3 tests/smoke_test.py --run-id <ID from running the previous command>
+"""
+
+
+class WILDSSmokeTestSuite:
+    # Only use 10% of the data splits for testing purposes
+    DATASET_FRACTION = 0.1
```

Comment:
Wonder if we can reduce this even more if we're just checking for reproducibility.

Reply:
I guess it depends on whether we're checking for exactness (down to random seed variation) vs. general correctness (results that roughly match what we reported, up to CIs).